src/runtime/propertyobject.cs

```diff
@@ -160,5 +198,54 @@ public static IntPtr tp_repr(IntPtr ob)
             var self = (PropertyObject)GetManagedObject(ob);
             return Runtime.PyString_FromString($"<property '{self.info.Name}'>");
         }
+
+        private static Func<object, object> BuildGetter(PropertyInfo propertyInfo)
+        {
+            var methodInfo = propertyInfo.GetGetMethod();
+            if (methodInfo == null)
+            {
+                // if the getter is not public 'GetGetMethod' will not find it
```

Comment:
According to [MSDN](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.propertyinfo.getgetmethod?view=netframework-4.0#System_Reflection_PropertyInfo_GetGetMethod_System_Boolean_), `GetGetMethod` has an overload, that returns non-public getters, so this condition can be removed, along with `getterCacheFailed`. Same for the setters I assume